src/odb/include/odb/dbStream.h

```diff
@@ -74,100 +69,67 @@ class dbOStream
 
   dbOStream& operator<<(char c)
   {
-    int n = fwrite(&c, sizeof(c), 1, _f);
-    if (n != 1) {
-      write_error();
-    }
+    _f << c;
     return *this;
   }
 
   dbOStream& operator<<(unsigned char c)
   {
-    int n = fwrite(&c, sizeof(c), 1, _f);
-    if (n != 1) {
-      write_error();
-    }
+    _f << c;
     return *this;
   }
 
   dbOStream& operator<<(int16_t c)
   {
-    int n = fwrite(&c, sizeof(c), 1, _f);
-    if (n != 1) {
-      write_error();
-    }
+    _f.write(reinterpret_cast<char*>(&c), sizeof(c));
```

Comment:
why doesn't _f << c work here?

Reply:
I think if the file is open in text mode it will potentially write out string(c) instead of bytes(c). Could be mistaken on the behavior of 16 bit values, but I think 32 and 64 bit values have that behavior.

Reply:
Why not use binary mode?  This isn't really a text file.

Reply:
The dbOstream constructor takes a generic stream std::ostream which we don't know anything about. If we're passed a ofstream in text mode we wouldn't know it. My C++ isn't that great though so my understanding of the situation is if that happens and we _f << uint16_t it will literally write out the string representation of that number whereas if we use the write above no matter the stream we're passed it will always write out the byte representation of the number. 

To your understand of the stdlib is that correct or am I wrong.

Reply:
On further research it appears your code is correct.

Reply:
Can we break out a function with generics here where we can document what is going on. Seems like it is worth doing so, there is something new and subtle to learn here for future readers of the code...